utils.py

```diff
@@ -40,3 +44,180 @@ def to_snake(name):
     Convert plugin name from snake case to dash representation
     """
     return name.replace("-", "_")
+
+
+# Changelog to markdown migration
+
+
+def migrate_changelog_to_markdown(plugin_root: str):
+    """
+    - Converts the changelog if there is a CHANGES.rst
+    - Updates the pyproject.toml "tool.towncrier" section to use markdown if it doesnt.
+    """
+    root_path = Path(plugin_root)
+    plugin_name = root_path.stem
+
+    # pyproject update
+    pyproject_file = root_path / "pyproject.toml"
+    if (
+        r'start_string = "[//]: # (towncrier release notes start)\n"'
+        not in pyproject_file.read_text()
+    ):
+        update_pyproject_toml(pyproject_file)
+
+    # changelog conversion
+    rst_changes = root_path / "CHANGES.rst"
+    if rst_changes.exists():
+        convert_changelog(rst_changes, plugin_name)
+        rst_changes.unlink()
+
+
+def update_pyproject_toml(pyproject_file: Path) -> Path:
+    import tomlkit
+
+    pyproject_data = tomlkit.loads(pyproject_file.read_text())
+
+    package_name = pyproject_data["tool"]["towncrier"]["package"]  # type: ignore
+    TOWNCRIER_TOML_DATA = {
+        "package": package_name,
+        "filename": "CHANGES.md",
+        "directory": "CHANGES/",
+        "title_format": "## {version} ({project_date}) {{: #{version} }}",
+        "template": "CHANGES/.TEMPLATE.md",
+        "issue_format": Template(
+            "[#{issue}](https://github.com/pulp/$plugin_name/issues/{issue})"
+        ).substitute(plugin_name=package_name),
+        "start_string": "[//]: # (towncrier release notes start)\n",
+        "underlines": ["", "", ""],
+    }
+    # update towncrier section
+    for k, v in TOWNCRIER_TOML_DATA.items():
+        pyproject_data["tool"]["towncrier"][k] = v  # type: ignore
+
+    # re-write file
+    pyproject_file.write_text(tomlkit.dumps(pyproject_data))
+    return pyproject_file
+
+
+def convert_changelog(changes_rst: Path, plugin_name: str) -> Path:
+    """Convert an rst changelog file to markdown.
+
+    Args:
+        changelog_file: The path to a rst changelog.
+    Return:
+        Path to new markdown changelog.
+    """
+    import pypandoc
+
+    # pre-process
+    with open(changes_rst, "r") as f:
```

Comment:
Why are you mixing pathlib and open usage in the same file?

Reply:
No particular reason, I'll fix that.